crypto/cloudkms/cloudkms.go

```diff
@@ -57,6 +60,25 @@ func (k Key) ResourceID() string {
 	)
 }
 
+func KeyFromResourceID(resourceID string) (Key, error) {
+	key := Key{}
+
+	scanned, err := fmt.Sscanf(
+		strings.ReplaceAll(resourceID, "/", " "),       // input
+		strings.ReplaceAll(resourceIDFormat, "/", " "), // format
+		&key.ProjectID, &key.LocationID, &key.KeyRingID, &key.KeyID, &key.KeyVersion, // arugments to fill
+	)
+
+	if err != nil {
+		return key, fmt.Errorf("cloudkms: failed to parse resource ID %s, scanf error: %w", resourceID, err)
+	}
+	if scanned != 5 {
+		return key, fmt.Errorf("cloudkms: failed to parse resource ID %s, found %d arguments but expected 5", resourceID, scanned)
```

Comment:
Consider refactoring the magic number in both the condition and the error message to a constant